lighthouse-core/lib/url-shim.js

```diff
@@ -128,4 +128,13 @@ URL.equalWithExcludedFragments = function(url1, url2) {
   }
 };
 
+/**
+ * Returns protocol of the given URL
+ * @param {string} url
+ * @return {string}
+ */
+URL.getProtocol = function(url) {
+  return new URL(url).protocol;
```

Comment:
generally we've only been adding these methods to this shim if it needs a built-in try/catch. Otherwise maybe just use `new URL(url).protocol` directly in `uses-request-compression.js`